packages/hooks/use-attrs/index.ts

```diff
@@ -9,29 +11,26 @@ interface Params {
 const DEFAULT_EXCLUDE_KEYS = ['class', 'style']
 const LISTENER_PREFIX = /^on[A-Z]/
 
-export default (params: Params = {}) => {
+export default (params: Params = {}): ComputedRef<Record<string, unknown>> => {
   const { excludeListeners = false, excludeKeys = [] } = params
-  const instance = getCurrentInstance()
-  const attrs = shallowRef({})
   const allExcludeKeys = excludeKeys.concat(DEFAULT_EXCLUDE_KEYS)
 
-  // Since attrs are not reactive, make it reactive instead of doing in `onUpdated` hook for better performance
-  instance.attrs = reactive(instance.attrs)
-
-  watchEffect(() => {
-    const res = entries(instance.attrs).reduce((acm, [key, val]) => {
-      if (
-        !allExcludeKeys.includes(key) &&
-        !(excludeListeners && LISTENER_PREFIX.test(key))
-      ) {
-        acm[key] = val
-      }
-
-      return acm
-    }, {})
-
-    attrs.value = res
-  })
-
-  return attrs
+  const instance = getCurrentInstance()
+  if (!instance) {
+    warn(
+      'use-attrs',
+      'getCurrentInstance() returned null. useAttrs() must be called at the top of a setup function'
+    )
+    return computed(() => ({}))
+  }
+
+  return computed(() =>
+    Object.fromEntries(
+      Object.entries(instance.proxy?.$attrs).filter(
```

Comment:
I think if you don't use reactive to make this `$attrs` reactive, it should be a plain object with no reactivity cc @SimonaliaChen for comfirmation.

Reply:
`instance.$proxy` is reactived